dumper/lib-common/src/main/java/com/google/edwmigration/dumper/plugin/ext/jdk/progress/RecordProgressMonitor.java

```diff
@@ -44,6 +46,7 @@ public class RecordProgressMonitor extends AbstractProgressMonitor {
   private long nextMinSeconds = 0;
 
   public RecordProgressMonitor(@Nonnull String name, @Nonnegative long total) {
+    checkArgument(total >= 0, "Total must be nonnegative. Was: %s", total);
```

Comment:
This is also the question about applying this annotation to the parameter of the `AbstractJdbcTask#newCsvResultSetExtractor` method in this PR.

Reply:
I expected it to work the same way as `@Nonnull`, i.e. cause a warning when a violation of the constraint is detected by static analysis. I used it to replace `@CheckForSigned`, because a negative parameter is no longer valid in this method.

After some local checks, I see that - at least in my case - the annotation doesn't seem to do any checks. It can still have some use for documenting the code, but the main purpose isn't fulfilled, so I can remove it to simplify the declarations (both usages).

Reply:
Please don't use annotations that don't work. The `checkArgument` call is enough to represent that the value is non-negative. I don't think additional documenting is needed, so I would vote for removing the annotation.

Reply:
Done, I removed all `@Nonnegative` annotations that would be added by this PR.